submissions/Nemooochka/js-dom/js/script.js

```diff
@@ -0,0 +1,133 @@
+        'description': 'The true story of how the Boston Globe uncovered the massive scandal of child molestation and cover-up within the local Catholic Archdiocese, shaking the entire Catholic Church to its core.',
+        'director': 'Tom McCarthy',
+        'writers': 'Josh Singer, Tom McCarthy',
+        'stars': 'Mark Ruffalo, Michael Keaton, Rachel McAdams'
+    },
+    {
+        'year': '2014',
+        'name': 'Birdman',
+        'poster': 'https://m.media-amazon.com/images/M/MV5BODAzNDMxMzAxOV5BMl5BanBnXkFtZTgwMDMxMjA4MjE@._V1_UX182_CR0,0,182,268_AL_.jpg',
+        'description': 'A washed-up superhero actor attempts to revive his fading career by writing, directing, and starring in a Broadway production.',
+        'director': 'Alejandro G. Iñárritu',
+        'writers': 'Alejandro G. Iñárritu, Nicolás Giacobone',
+        'stars': 'Michael Keaton, Zach Galifianakis, Edward Norton'
+    }
+];
+
+const initialMovie = moviesData[0];
+const sidebar = document.querySelector(".sidebar");
+const sidebarNav = document.querySelector(".sidebar-nav");
+const menuChevron = document.querySelector(".menu-chevron");
+
+const toggleSidebar = () => {
+    sidebar.classList.toggle('open');
+};
+
+function generateView({poster, name, description, director, writers, stars}) {
+    return `<div class="card-poster">
+                <img src="${poster}" alt="${name}">
+            </div>
+            <div class="card-info">
+                <h2>${name}</h2>
+                <div class="details">
+                    <p class="details_description">${description}</p>
+                    <div class="details_director">
+                        <b>Director:</b>
+                        <span class="director">${director}</span>
+                    </div>
+                    <div class="details_writer">
+                        <b>Writers:</b>
+                        <span>${writers}</span>
+                    </div>
+                    <div class="details_stars">
+                        <b>Stars:</b>
+                        <span>${stars}</span>
+                    </div>
+                </div>
+            </div>`;
+}
+
+function renderCard(nodeList) {
+    const cardMovie = document.querySelector('.card');
+    cardMovie.innerHTML = generateView(nodeList);
+}
+
+function updateActiveNav(targetLink) {
+    const activeElement = document.querySelector('.sidebar-nav_link.active');
+    if (activeElement) {
+        activeElement.classList.remove('active');
+    }
+    targetLink.closest('.sidebar-nav_link').classList.add('active');
+}
+
+function changeMovie(e) {
+
+    let clickedElem = e.target;
+
+    if(clickedElem.tagName === 'A') {
+        e.preventDefault();
+
+        updateActiveNav(clickedElem);
+
+        let chosenYear = clickedElem.dataset.year;
+
+        let chosenMovie = moviesData.find(movie => {
+            for (let item in movie) {
```

Comment:
no need to use the `for` loop here. `Array.prototype.find` is already uses loop
```suggestion
            const chosenMovie = moviesData.find(movie => movie['year'] === chosenYear)
```